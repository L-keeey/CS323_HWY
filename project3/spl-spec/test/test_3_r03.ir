FUNCTION main :
t1 := #30
v1 := t1
t2 := #0
v2 := t2
t3 := #1
v3 := t3
t4 := #1
v4 := t4
LABEL label1 :
t5 := v4
t6 := v1
IF t5 <= t6 GOTO label2
GOTO label3
LABEL label2 :
LABEL label4 :
t7 := v3
t8 := v4
IF t7 <= t8 GOTO label5
GOTO label6
LABEL label5 :
t9 := v4
t13 := v4
t14 := v3
t11 := t13 / t14
t12 := v3
t10 := t11 * t12
IF t9 == t10 GOTO label7
GOTO label8
LABEL label7 :
t17 := v2
t18 := #1
t16 := t17 + t18
v2 := t16
t15 := v2
LABEL label8 :
t21 := v3
t22 := #1
t20 := t21 + t22
v3 := t20
t19 := v3
GOTO label4
LABEL label6 :
t23 := v2
t24 := #2
IF t23 == t24 GOTO label9
GOTO label10
LABEL label9 :
t25 := v4
WRITE t25
LABEL label10 :
t27 := #1
v3 := t27
t26 := v3
t29 := #0
v2 := t29
t28 := v2
t32 := v4
t33 := #1
t31 := t32 + t33
v4 := t31
t30 := v4
GOTO label1
LABEL label3 :
t34 := #0
RETURN t34
