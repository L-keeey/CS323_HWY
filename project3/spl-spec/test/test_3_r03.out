Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: N
                ASSIGN
                Exp (2)
                  INT: 30
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: num
                  ASSIGN
                  Exp (3)
                    INT: 0
                COMMA
                DecList (3)
                  Dec (3)
                    VarDec (3)
                      ID: i
                    ASSIGN
                    Exp (3)
                      INT: 1
                  COMMA
                  DecList (3)
                    Dec (3)
                      VarDec (3)
                        ID: k
                      ASSIGN
                      Exp (3)
                        INT: 1
              SEMI
        StmtList (4)
          Stmt (4)
            WHILE
            LP
            Exp (4)
              Exp (4)
                ID: k
              LE
              Exp (4)
                ID: N
            RP
            Stmt (4)
              CompSt (4)
                LC
                StmtList (5)
                  Stmt (5)
                    WHILE
                    LP
                    Exp (5)
                      Exp (5)
                        ID: i
                      LE
                      Exp (5)
                        ID: k
                    RP
                    Stmt (5)
                      CompSt (5)
                        LC
                        StmtList (6)
                          Stmt (6)
                            IF
                            LP
                            Exp (6)
                              Exp (6)
                                ID: k
                              EQ
                              Exp (6)
                                LP
                                Exp (6)
                                  Exp (6)
                                    Exp (6)
                                      ID: k
                                    DIV
                                    Exp (6)
                                      ID: i
                                  MUL
                                  Exp (6)
                                    ID: i
                                RP
                            RP
                            Stmt (7)
                              Exp (7)
                                Exp (7)
                                  ID: num
                                ASSIGN
                                Exp (7)
                                  Exp (7)
                                    ID: num
                                  PLUS
                                  Exp (7)
                                    INT: 1
                              SEMI
                          StmtList (8)
                            Stmt (8)
                              Exp (8)
                                Exp (8)
                                  ID: i
                                ASSIGN
                                Exp (8)
                                  Exp (8)
                                    ID: i
                                  PLUS
                                  Exp (8)
                                    INT: 1
                              SEMI
                        RC
                  StmtList (10)
                    Stmt (10)
                      IF
                      LP
                      Exp (10)
                        Exp (10)
                          ID: num
                        EQ
                        Exp (10)
                          INT: 2
                      RP
                      Stmt (11)
                        Exp (11)
                          ID: write
                          LP
                          Args (11)
                            Exp (11)
                              ID: k
                          RP
                        SEMI
                    StmtList (12)
                      Stmt (12)
                        Exp (12)
                          Exp (12)
                            ID: i
                          ASSIGN
                          Exp (12)
                            INT: 1
                        SEMI
                      StmtList (13)
                        Stmt (13)
                          Exp (13)
                            Exp (13)
                              ID: num
                            ASSIGN
                            Exp (13)
                              INT: 0
                          SEMI
                        StmtList (14)
                          Stmt (14)
                            Exp (14)
                              Exp (14)
                                ID: k
                              ASSIGN
                              Exp (14)
                                Exp (14)
                                  ID: k
                                PLUS
                                Exp (14)
                                  INT: 1
                            SEMI
                RC
          StmtList (16)
            Stmt (16)
              RETURN
              Exp (16)
                INT: 0
              SEMI
        RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION main :
ComSt
DefList
Def
DecList
Dec
Exp
const
INT
0
t1 := #30
v1 := t1
DefList
Def
DecList
Dec
Exp
const
INT
0
t2 := #0
